parser_library/src/lsp/completion_item.cpp

```diff
@@ -161,59 +162,54 @@ void process_mnemonic_code(
                 continue;
             }
         }
-        // do not replace by a mnemonic
-        std::string curr_op_with_mnem = "";
-        std::string curr_op_without_mnem = "";
-        if (no_optional == 0)
-        {
-            if (i != 0)
-                curr_op_with_mnem += ",";
-            if (!first)
-                curr_op_without_mnem += ",";
-            curr_op_with_mnem += mach_operands[i].to_string();
-            curr_op_without_mnem += mach_operands[i].to_string();
-        }
-        else if (no_optional == 1 && mach_operands.size() - i == 1)
+        const bool is_optional = mach_operands.size() - i <= no_optional;
```

Comment:
I know it's not the scope of this PR but how do you feel about renaming `no_optional` to something little bit different? E.g. `optional_op_count`, `optional_count` or similar?

I've got confused for a solid minute why there is `is_optional` and `no_optional` while having different data types